wps-common/src/main/java/au/org/aodn/aws/wps/status/StatusHelper.java

```diff
@@ -1,20 +1,179 @@
+    {
+        return WpsConfig.getS3BaseUrl() + s3Bucket + "/" + S3Key;
+    }
+
+
+    /**
+     * Generate an XML string from a response XML type.
+     * @param response
+     * @return
+     */
+    public static String createResponseXmlDocument (ResponseBaseType response) {
+        String responseDoc = null;
+
+        JAXBContext context;
+        try {
+            context = JAXBContext.newInstance(response.getClass());
+            Marshaller m = context.createMarshaller();
+            m.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+            StringWriter writer = new StringWriter();
+            m.marshal(response, writer);
+            responseDoc = writer.toString();
+
+        } catch (JAXBException e) {
+            e.printStackTrace();
+        }
+        return responseDoc;
+    }
+
+
+    public static ProcessFailedType getProcessFailedType(String message, String code)
+    {
+        ProcessFailedType failed = new ProcessFailedType();
+        failed.setExceptionReport(getExceptionReport(message, code));
+        return failed;
+    }
+
+
+
+    public static ExceptionReport getExceptionReport(String message, String code)
+    {
+        ExceptionReport report = new ExceptionReport();
+        ExceptionType type = new ExceptionType();
+        type.getExceptionText().add(message);
+        type.setExceptionCode(code);
+        report.getException().add(type);
+        return report;
+    }
+
+
+    public static String getExceptionReportString(String message, String code)
+    {
+        ExceptionReport report = getExceptionReport(message, code);
+        String responseDoc = null;
+
+        JAXBContext context;
+        try {
+            context = JAXBContext.newInstance(ExceptionReport.class);
+            Marshaller m = context.createMarshaller();
+            m.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+            StringWriter writer = new StringWriter();
+            m.marshal(report, writer);
+            responseDoc = writer.toString();
+
+        } catch (JAXBException e) {
+            e.printStackTrace();
         }
+        return responseDoc;
+    }
+
+
+    public static ProcessStartedType getProcessStartedType(String message, Integer percentComplete)
+    {
+        ProcessStartedType started = new ProcessStartedType();
+        started.setValue(message);
+        started.setPercentCompleted(percentComplete);
+        return started;
+    }
+
+
+    public static final XMLGregorianCalendar getCreationDate()
+            throws DatatypeConfigurationException
+    {
+        GregorianCalendar currentTime = new GregorianCalendar();
+        currentTime.setTime(new Date());
+        currentTime.setTimeZone(TimeZone.getTimeZone("GMT"));
+        return DatatypeFactory.newInstance().newXMLGregorianCalendar(currentTime);
+    }
 
-        return statusDocument;
+
+    public static void addExecuteOutputReference(ExecuteResponse response, String outputIdentifier, String outputHref)
```

Comment:
Will this be useful outside of execute status documents? If not it should be moved into the ExecuteStatusBuilder class as a private method.